internal/daysteps/daysteps.go

```diff
@@ -1,13 +1,129 @@
 package daysteps
 
+import (
+    "fmt"
+    "strings"
+    "time"
+	"strconv"
+    
+    "github.com/Yandex-Practicum/tracker/internal/personaldata"
+	"github.com/Yandex-Practicum/tracker/internal/spentenergy"
+)
 type DaySteps struct {
-	// TODO: добавить поля
+    Steps    int
+    Duration time.Duration
+	personaldata.Personal
 }
 
 func (ds *DaySteps) Parse(datastring string) (err error) {
-	// TODO: реализовать функцию
+    if len(datastring) < 7 {
+        return fmt.Errorf("некорректная длина строки")
+    }
+    
+    parts := strings.Split(datastring, ",")
+    if len(parts) != 2 {
+        return fmt.Errorf("неверный формат данных: ожидается 2 поля, получено %d", len(parts))
+    }
+    
+    steps, err := strconv.Atoi(parts[0])
+    if err != nil {
+        return fmt.Errorf("ошибка парсинга количества шагов: %w", err)
+    }
+    if steps <= 0 {
+        return fmt.Errorf("количество шагов должно быть положительным")
+    }
+    ds.Steps = steps
+    
+    durationStr := parts[1]
+    var hours, minutes float64
+    
+    if strings.Contains(durationStr, "h") && strings.Contains(durationStr, "m") {
```

Comment:
Пакет time отлично все парсит
```suggestion
    ds.Duration, err = time.ParseDuration(data[1])
```